Chirpy theme: ‘a’ tag is missing a reference when author url is missing

### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Describe the bug

 GH Actions build jobs are failing when author url is missing.

see build example here https://github.com/SRodi/srodi.github.io/actions/runs/7058684466/job/19214755311

```
Running 3 checks (Scripts, Links, Images) in ["_site"] on *.html files...


Checking 43 internal links
Checking internal link hashes in 2 files
Ran on 17 files!


For the Links check, the following failures were found:

* At _site/posts/how-to-test-terraform-modules/index.html:1:

  'a' tag is missing a reference


HTML-Proofer found 1 failure!
Error: Process completed with exit code 1.
```

### Steps To Reproduce

Assuming `GitHub actions` are configured to automatically build the project on commit pushes to remote.

1. Create file `_data/_authors.yml` with the author info omitting the url, something like 

```yml
srodi:
  name: Simone Rodigari
```

2. Add author details in the _Front Matter_ as follows

```yml
---
title: "..."
date: 2023-09-19 09:00:00 +0100
categories: ["...", "..."]
tags: ["...", "..."]
author: srodi
...
---
```

3. GH Action build job fails

### Expected Behavior

* job should not fail
* error message could be more descriptive
* default url could be set? (i.e. website url)

### Environment

```
❯ ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-darwin23]
❯ bundle exec jekyll -v
jekyll 4.3.2
❯ bundle info jekyll-theme-chirpy
  * jekyll-theme-chirpy (6.3.1)
        Summary: A minimal, responsive, and feature-rich Jekyll theme for technical writing.
        Homepage: https://github.com/cotes2020/jekyll-theme-chirpy
        Documentation: https://github.com/cotes2020/jekyll-theme-chirpy/#readme
        Source Code: https://github.com/cotes2020/jekyll-theme-chirpy
        Wiki: https://github.com/cotes2020/jekyll-theme-chirpy/wiki
        Bug Tracker: https://github.com/cotes2020/jekyll-theme-chirpy/issues
        Path: /Users/simone/.gem/ruby/3.2.2/gems/jekyll-theme-chirpy-6.3.1
```


### Anything else?

Reference to community blog post with solution: https://talk.jekyllrb.com/t/chirpy-theme-a-tag-is-missing-a-reference/8731

--- a/_layouts/post.html
+++ b/_layouts/post.html
@@ -61,7 +61,11 @@ tail_includes:
           <em>
             {% if authors %}
               {% for author in authors %}
-                <a href="{{ site.data.authors[author].url }}">{{ site.data.authors[author].name }}</a>
+                {% if site.data.authors[author].url -%}
+                  <a href="{{ site.data.authors[author].url }}">{{ site.data.authors[author].name }}</a>
+                {%- else -%}
+                  {{ site.data.authors[author].name }}
+                {%- endif %}
                 {% unless forloop.last %}{{ '</em>, <em>' }}{% endunless %}
               {% endfor %}
             {% else %}

